mysite/ct/templatetags/ct_extras.py

```diff
@@ -211,3 +211,12 @@ def singularize(i, s):
     return '%s %d, %d' % (monthStrings[dt.month - 1], dt.day, dt.year)
 
 
+@register.filter
+def filter_input(edge, unit_lesson):
+    """Filter input UnitLesson to be acceptable for this edge.
+
+    :param edge: FSMEdge
+    :param unit_lesson: UnitLesson
+    :return:
+    """
+    return edge.fromNode._plugin.select_UnitLesson_filter(unit_lesson)
```

Comment:
Nice implementation.  I think this can be generalized a bit:

* add a general method `FSMEdge.filter_input(self, obj), something like:

    def filter_input(self, obj):
        try: # see if node intercepts this select_X_filter() call
            func = getattr(self.fromNode._plugin, self.name + '_filter')
        except AttributeError: # not intercepted, so accept by default
            return True
        else:
            return func(obj)

* then the `filter_input(edge, obj)` code just becomes `return edge.filter_input(obj)`